includes/class-wc-payments-payment-method-messaging-element.php

```diff
@@ -65,8 +79,10 @@ public function init(): string {
 			'WCPAY_PRODUCT_DETAILS',
 			'wcpayStripeSiteMessaging',
 			[
-				'price'          => WC_Payments_Utils::prepare_amount( $price, $currency_code ),
+				'price'          => $base_product_price,
```

Comment:
Let's replace `price` with `'productId' => 'base_product'`. A bit more JS tweaks, and we should be good.

Reply:
Is this because we are already sending the base product price within `productPrices` and to avoid duplication?

Reply:
Yep! And the `productId` will track which is selected on JS side (no selectedVariationPrice is needed, just keep productId updated and get a price based on what's in productId). 

The targeted implementation will be pretty much readable.

Reply:
That was a brilliant pointer. Thanks! 

I have implemented it here - [fb812a0](https://github.com/Automattic/woocommerce-payments/pull/6626/commits/fb812a059a7fc77f0f53ccf6d34e8a2d0dffa6a4)

I hope I have done it correctly.